c7n/resources/rds.py

```diff
@@ -1935,3 +1935,43 @@ def process(self, resources, event=None):
                 r['c7n:Engine'] = v
                 matched.append(r)
         return matched
+
+
+@filters.register('db-option-groups')
+class DbOptionGroups(ValueFilter):
+    """This filter describes RDS option groups for associated RDS instances.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: rds-data-in-transit-encrypted
+            resource: aws.rds
+            filters:
+              - type: db-option-groups
+              - type: value
+                key: Options
+                value: not-null
+              - type: value
+                key: Options[].OptionName
+                op: intersect
+                value:
+                  - NATIVE_NETWORK_ENCRYPTION
+                  - SSL
+    """
+
+    schema = type_schema('db-option-groups', rinherit=ValueFilter.schema)
+    permissions = ('rds:DescribeDBInstances', 'rds:DescribeOptionGroups', )
+
+    def process(self, resources, event=None):
+        results = []
+        client = local_session(self.manager.session_factory).client('rds')
+        for r in resources:
+            option_group_name = r['OptionGroupMemberships'][0]['OptionGroupName']
+            paginator = client.get_paginator('describe_option_groups')
+            response = paginator.paginate(
+                OptionGroupName=option_group_name).build_full_result().get('OptionGroupsList', [])
+            for pg in response:
+                results.append(pg)
+        return results
```

Comment:
we don't really do this, its basically swapping out the policy specified resource to return something different.

typically we would have the options cached, and type: db-option-groups. ideally this would similar to how we handle db-parameter filter.

Reply:
@sgungi-intuit Problem with this PR is that you're introducing a filter and yet you don't do any filtering. You're just appending all of the option groups to `results`.

Anyway, as what Kapil has mentioned, you should do something similar to db-parameter filter https://github.com/cloud-custodian/cloud-custodian/blob/master/c7n/resources/rds.py#L1588

Reply:
@darrendao @kapilt, let me take db-parameter filter as reference and make the changes.

Reply:
updated code to use cache.